src/mame/sinclair/scorpion.cpp

```diff
@@ -168,7 +229,25 @@ INPUT_CHANGED_MEMBER(scorpion_state::on_nmi)
 	m_nmi_pending = newval;
 }
 
-void scorpion_state::port_7ffd_w(offs_t offset, u8 data)
+void scorpion_state::do_nmi()
+{
+	m_beta->enable();
+	scorpion_update_memory();
+	m_maincpu->pulse_input_line(INPUT_LINE_NMI, attotime::zero);
+	m_nmi_pending = 0;
+}
+
+u8 scorpion_state::port_ff_r()
+{
+	if (m_magic_lock)
+	{
+		LOGIO("Registers lock released\n");
+		m_magic_lock = 0;
+	}
```

Comment:
You need to add ` && !machine().side_effects_disabled()` to the condition here, or it will screw up when a memory window is open.